services/vision/obstaclesdepth/obstacles_depth.go

```diff
@@ -347,3 +332,28 @@ func (o *obsDepth) performKMeans3D(k int) ([]spatialmath.Geometry, clusters.Clus
 	}
 	return boxes, clusters, err
 }
+
+// makePointList will populate o.depthPts with the depth data.
+func (o *obsDepth) makePointList() [][]float64 {
+	width, height := o.dm.Width(), o.dm.Height()
+	out := make([][]float64, 0, width*height)
+	for i := 0; i < width; i += sampleN {
+		for j := 0; j < height; j++ {
+			out = append(out, []float64{float64(i), float64(j), float64(o.dm.GetDepth(i, j))})
+		}
+	}
+	o.depthPts = out
+	return out
+}
+
+func (o *obsDepth) isObstaclePoint(candidate []float64) bool {
+	for _, pt := range o.depthPts {
+		if candidate[0] == pt[0] && candidate[1] == pt[1] {
+			continue
+		}
+		if o.isCompatible(candidate, pt) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
everything in this file looks good but I'm confused about the relevance to this PR. maybe it's obvious but I'm missing something

Reply:
Nice catch - didn't rebase on main